ga4gh/datamodel/reads.py

```diff
@@ -75,3 +89,59 @@ def toProtocolElement(self):
         readGroup.referenceSetId = None
         readGroup.sampleId = None
         return readGroup
+
+    def getReadAlignments(self, referenceName, referenceId, start, end):
+        """
+        Returns an iterator over the specified reads
+        """
+        if referenceName is not None and referenceId is not None:
+            raise exceptions.BadReadsSearchRequestBothRefs()
+        if referenceId is not None:
+            referenceName = self._samFile.getrname(referenceId)
+        try:
+            readAlignments = self._samFile.fetch(referenceName, start, end)
+        except ValueError as exception:
```

Comment:
I'm not sure that this `try` block will work as expected. I think `readAlignments` is an iterator, so we may well throw the low-level errors from the for-loop below, when we actually do the iteration. Dealing with errors coming from pysam/htslib is going to be a challenge, so it might be better to just leave this try-except out for now.